lib/logstash/inputs/couchdb_changes.rb

```diff
@@ -0,0 +1,217 @@
+    @path         = '/' + @db + '/_changes'
+
+    @scheme = @secure ? 'https' : 'http'
+
+    @since = @initial_sequence ? @initial_sequence : @sincedb.read
+
+    if !@username.nil? && !@password.nil?
+      @userinfo = @username + ':' + @password
+    else
+      @userinfo = nil
+    end
+    
+  end
+  
+  module SinceDB
+    class File
+      def initialize(file)
+        @sincedb_path = file
+      end
+
+      def read
+        ::File.exists?(@sincedb_path) ? ::File.read(@sincedb_path).chomp.strip : 0
+      end
+
+      def write(since = nil)
+        since = 0 if since.nil?
+        ::File.write(@sincedb_path, since.to_s)
+      end
+    end
+  end
+  
+  public
+  def run(queue)
+    buffer = FileWatch::BufferedTokenizer.new
+    @logger.info("Connecting to CouchDB _changes stream at:", :host => @host.to_s, :port => @port.to_s, :db => @db)
+    uri = build_uri
+    Net::HTTP.start(@host, @port, :use_ssl => (@secure == true), :ca_file => @ca_file) do |http|
+      request = Net::HTTP::Get.new(uri.request_uri)
+      http.request request do |response|
+        raise ArgumentError, "Database not found!" if response.code == "404"
+        response.read_body do |chunk|
+          buffer.extract(chunk).each do |changes|
+            next if changes.chomp.empty?
+            event = build_event(changes)
+            @logger.debug("event", :event => event.to_hash_with_metadata) if @logger.debug?
+            decorate(event) if @decorate_event
+            unless event["empty"]
+              queue << event
+              @since = event['@metadata']['seq']
+              @sincedb.write(@since.to_s)
+            end
+          end
+        end
+      end
+    end
+  rescue Timeout::Error, Errno::EINVAL, Errno::ECONNRESET, EOFError, Errno::EHOSTUNREACH, Errno::ECONNREFUSED,
+    Net::HTTPBadResponse, Net::HTTPHeaderSyntaxError, Net::ProtocolError => e
+    @logger.error("Connection problem encountered: Retrying connection in 10 seconds...", :error => e.to_s)
+    retry if reconnect?
+  rescue Errno::EBADF => e
+    @logger.error("Connction refused due to bad file descriptor: ", :error => e.to_s)
+    retry if reconnect?
+  rescue ArgumentError => e
+    @logger.error("Unable to connect to database", :db => @db, :error => e.to_s)
+    retry if reconnect?
+  end
+  
+  private
+  def build_uri
+    options = {:feed => @feed, :include_docs => @include_docs, :since => @since}
+    options = options.merge(@timeout ? {:timeout => @timeout} : {:heartbeat => @heartbeat})
+    URI::HTTP.build(:scheme => @scheme, :userinfo => @userinfo, :host => @host, :port => @port, :path => @path, :query => URI.encode_www_form(options))
+  end
+
+  private
+  def reconnect?
+    sleep(@always_reconnect ? @reconnect_delay : 0)
+    @always_reconnect
+  end
+
+  private
+  def build_event(line)
+    # In lieu of a codec, build the event here
+    line = LogStash::Json.load(line)
+    return LogStash::Event.new({"empty" => true}) if line.has_key?("last_seq")
```

Comment:
why not just return `nil` from build_event instead of `LogStash::Event.new({"empty" => true}` and refactor the `run` method with something like this?

```ruby
            if event = build_event(changes)
              @logger.debug("event", :event => event.to_hash_with_metadata) if @logger.debug?
              decorate(event)
              queue << event
              @sequence = event['@metadata']['seq']
              @sequencedb.write(@sequence.to_s)
            end
```